lib/spack/spack/test/compilers.py

```diff
@@ -192,6 +197,98 @@ def try_all_dirs(*args):
     assert set(retrieved_rpaths) == expected_rpaths
 
 
+def test_get_compiler_link_paths(monkeypatch):
+    # Mock compiler call
+    test_dirs = ['/path/to/first/lib', '/path/to/second/lib64']
+    test_output = 'ld -L%s -L%s' % tuple(test_dirs)
+
+    def call_compiler(*args, **kwargs):
+        return test_output
+    monkeypatch.setattr(
+        spack.util.executable.Executable, '__call__', call_compiler)
+
+    compiler = MockCompiler()
```

Comment:
This code is repeated many times in the tests below with a couple of slight variations. Maybe we can code a fixture that aggregates that. 

Another non immediate thing is that the real test is, if I understand correctly, in the conditional of `call_compiler` instead of being in the assertion. The output is constructed statically to pass the assertion and the conditional is what returns the expected output or an empty list - if that's the case it would be clearer to assert on the condition in the `if` clause if possible.

Reply:
I disagree about where the test is. `call_compiler` has a conditional to emulate the behavior of an actual compiler that can change its linked libraries based on its flags. And the actual test is that it gets the version that it would expect to get.

I'll see what I can do about refactoring common code and push an update.

Reply:
> And the actual test is that it gets the version that it would expect to get.

Ok, makes sense. Thanks for the explanation. Maybe a comment somewhere in the tests could help future readers? Also is it worth to make this explicit and check that if you change flags in the same test you obtain the other result?